src/crimson/osd/backfill_state.cc

```diff
@@ -444,6 +465,15 @@ BackfillState::ReplicasScanning::react(ReplicaScanned evt)
   return discard_event();
 }
 
+boost::statechart::result
+BackfillState::ReplicasScanning::react(Cancel evt)
+{
+  logger().debug("{}: cancelled within ReplicasScanning",
+                 __func__);
+  waiting_on_backfill.clear();
```

Comment:
It seems that `BackfillState::progress_tracker` also need to be cleared, am I right?